app/src/main/java/com/android/sample/ui/recipeOverview/RecipeOverview.kt

```diff
@@ -0,0 +1,361 @@
+                    .verticalScroll(scrollState)) {
+              // Display of the recipe image
+              RecipeImage(currentRecipe, width, height)
+              // Display of the recipe title
+              RecipeDescription(currentRecipe)
+              Spacer(modifier = Modifier.size(16.dp))
+              // Display of the prepare, cook and the total time
+              PrepareCookTotalTimeDisplay()
+              Spacer(modifier = Modifier.size(14.dp))
+              // Display of the Ingredient and Instruction buttons that allow us to change between
+              // the different views
+              Column(
+                  horizontalAlignment = Alignment.CenterHorizontally,
+                  modifier =
+                      Modifier.background(
+                              MaterialTheme.colorScheme.primary, shape = RoundedCornerShape(5))
+                          .fillMaxSize()) {
+                    Spacer(modifier = Modifier.height(6.dp))
+                    // Display of the buttons
+                    Row {
+                      // Display of the Ingredients button
+                      Button(
+                          onClick = { ingredientsView = true },
+                          shape = RoundedCornerShape(0.dp),
+                          modifier = Modifier.width(150.dp).testTag("ingredientsButton"),
+                          colors =
+                              ButtonColors(goldenBronze, Color.Black, goldenBronze, Color.Black),
+                          border = BorderStroke(2.dp, Color.Black)) {
+                            Text(
+                                text = "Ingredients",
+                                fontSize = 12.sp,
+                                textAlign = TextAlign.Center)
+                          }
+                      // Display of the Instruction buttons
+                      Button(
+                          onClick = { ingredientsView = false },
+                          shape = RoundedCornerShape(0.dp),
+                          modifier = Modifier.width(150.dp).testTag("instructionsButton"),
+                          colors =
+                              ButtonColors(goldenBronze, Color.Black, goldenBronze, Color.Black),
+                          border = BorderStroke(2.dp, Color.Black)) {
+                            Text(
+                                text = "Instructions",
+                                fontSize = 12.sp,
+                                textAlign = TextAlign.Center)
+                          }
+                    }
+                    // Display of the list of ingredients with the ability to change the number of
+                    // servings
+                    IngredientInstructionView(ingredientsView, servingsCount, currentRecipe) {
+                        newServingsCount ->
+                      servingsCount = newServingsCount
+                    }
+                  }
+            }
+      }
+}
```

Comment:
We could extract some of the code in a function.

Reply:
Done!